consensus/state.go

```diff
@@ -354,6 +355,30 @@ func (cs *State) OnStart() error {
 		return err
 	}
 
+	// Double Signing Risk Reduction Logic
+	// triggered only when validator mode(existing privValidator) and DoubleSignCheckHeight set
+	if !(cs.privValidator == nil) && cs.config.DoubleSignCheckHeight > 0 && cs.Height > 0 {
+		valAddr, err := cs.privValidator.GetPubKey()
+		if err != nil {
+			return err
+		}
+
+		doubleSignCheckHeight := cs.config.DoubleSignCheckHeight
+		if doubleSignCheckHeight > cs.Height {
```

Comment:
shouldn't we compare with `cs.Height-1`?

Reply:
If that's the case, I expect this result ![10](https://tva1.sinaimg.cn/large/007S8ZIlgy1gh25hm2pb6j312z0u0drk.jpg)